Relations dto fix

SzBeni2003 opened the pull request.

martonbakk committed:
user entity fix

martonbakk committed:
event entity fix

SzBeni2003 committed:
Using Dtos - not yet working!!

SzBeni2003 committed:
Yarn.lock exterminated

mozsarmate committed:
modositas

mozsarmate committed:
Revert "modositas"

This reverts commit 33a73a68747f2cca9d4fa4fee0dd8b4b937cd5a3.

mozsarmate committed:
schema

mozsarmate committed:
migration deleted

mozsarmate committed:
create-event-dto fix

mozsarmate committed:
migrations

mozsarmate committed:
create user dto fix

SzBeni2003 committed:
use user-dto

SzBeni2003 committed:
.env-et nem találja

SzBeni2003 committed:
Backend (hopefully) ready for deployment

DannySS123 commented on apps/backend/src/event/entity/event.entity.ts:
```diff
@@ -1,7 +1,8 @@
 
-export enum Priority {
-  NONE = 'NONE',
+/*export enum Relevance {
```
Why is this commented out?

SzBeni2003 replied:
We overlooked issue #17 , and it haven't been cleaned up since we found the solution in the last meeting. Those enums were unnecessary. I will clean it up.

DannySS123 commented on apps/backend/src/users/users.service.ts:
```diff
@@ -26,11 +29,19 @@ export class UsersService {
+    try {
+      return await this.prisma.user.update({ where: { authSchId: id }, data });
+    } catch {
+      throw new NotFoundException(`User not found`);
```
Is this the only error we can get here? Can we get an error if the user is found, but the data s not correct?

If we can have more errors, we should throw different error massages based on the error type. For reference:
https://www.prisma.io/docs/orm/reference/error-reference

SzBeni2003 replied:
I believe that only realistic error we can get is NotFoundException, since the point of issue #15 was to validate all the data before it reached Prisma. Isn't that correct?

DannySS123 replied:
That's true, but can you please try the endpoints with Postmen to see if such errors are possible? And if not then it is good like this.

DannySS123 replied:
I think the email is unique, so it can give a unique constraint violation. Btw we shouldn't allow the user to change their name, it should be the same as the one from autsch.

DannySS123 reviewed (commented):
Great job!! 

Please take a look at the different errors Prisma can give

SzBeni2003 committed:
code cleanup

SzBeni2003 committed:
added isemail validator to email property

DannySS123 commented:
@SzBeni2003 Good, but this validator won't solve the problem, in the user update function, in the catch part, please catch the error, and based on its type throw an error.
Use this for reference: https://www.prisma.io/docs/orm/reference/error-reference

SzBeni2003 committed:
update method throws error based on error type

DannySS123 commented on apps/backend/src/event/event.service.ts:
```diff
@@ -26,11 +29,16 @@ export class EventService {
-    } catch {
-      throw new NotFoundException(`Event not found`);
+    } catch (error) {
+      if (error instanceof NotFoundException) {
```
I don't think it will work like this, we should use the `PrismaClientKnownRequestError` since it contains the code of the error, from which we can decide what kind of error to throw.  Check the error codes in the Prisma docs (I already sent some links)
Here is an example from Konzisite:
![image](https://github.com/user-attachments/assets/494aef4f-eaed-4e94-87fb-528ca65dacc5)

SzBeni2003 replied:
Just to make sure: P2001 should be the code if there's no event with the id, and P2006 should be the code if the data is invalid, right?
I'm not sure whether it is P2006 or P2007, but this looks more logical

DannySS123 replied:
Hmm not sure, but please try out both cases and if you check the error code, you will see what kind of errors we get.

SzBeni2003 committed:
Endpoints checked
invalid data: PrismaClientValidationError,
not found: PrismaClientKnownRequestError

SzBeni2003 committed:
shutted up prettier
